customer/api.go

```diff
@@ -0,0 +1,178 @@
+package customer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type API struct {
+	clientID   string
+	license    int32
+	HTTPClient *http.Client
+	ApiURL     string
+	token      string
+	tokenLock  sync.Mutex
+}
+
+func NewAPI(clientID string, license int32) *API {
+	return &API{
+		clientID:   clientID,
+		license:    license,
+		ApiURL:     "https://api.livechatinc.com/",
+		HTTPClient: http.DefaultClient,
+	}
+}
+
+// SetToken sets customer token. This is required for requests authorization.
+func (a *API) SetToken(t string) {
+	a.tokenLock.Lock()
+	defer a.tokenLock.Unlock()
+
+	a.token = t
+}
+
+type continuousChat struct {
+	*Chat
+	Continuous bool `json:"continuous"`
+}
+
+func (a *API) StartChat(c *Chat, continuous bool) (chatID, threadID string, err error) {
+	if c.ID != "" {
+		return "", "", fmt.Errorf("chat %s already started", c.ID)
+	}
+
+	cc := continuousChat{c, continuous}
+	body, err := a.call("start_chat", cc)
+
+	if err != nil {
+		return "", "", err
+	}
+
+	resp := struct {
```

Comment:
I think it'd be nice to move this struct to some `internal` file because it'll be reused in agent-api client

Reply:
Also applies to other places where raw `struct` is used.